content/en/docs/tasks/manage-kubernetes-objects/declarative-config.md

```diff
@@ -371,44 +371,81 @@ to result in the user deleting something unintentionally:
 kubectl delete -f <filename>
 ```
 
-### Alternative: `kubectl apply -f <directory/> --prune -l your=label`
+### Alternative: `kubectl apply -f <directory/> --prune`
 
-Only use this if you know what you are doing.
+As an alternative to `kubectl delete`, you can use `kubectl apply` to identify objects to be deleted after their configuration files have been removed from the directory.
 
-{{< warning >}}
-`kubectl apply --prune` is in alpha, and backwards incompatible
-changes might be introduced in subsequent releases.
-{{< /warning >}}
+There are currently two pruning modes available in kubectl apply:
+- Allowlist-based pruning: This mode has existed since kubectl v1.5 but is still in alpha due to usability, correctness and performance issues with its design. The ApplySet-based mode is designed to replace it, and upon its graduation, allowlist-based pruning will be deprecated using beta timelines.
+- ApplySet-based pruning: An ApplySet is a server-side object (by default, a Secret) that kubectl can use to accurately and efficiently track set membership across `apply` operations. This mode was introduced in kubectl v1.27 as a replacement for allowlist-based pruning. It is under active development.
```

Comment:
Is this alpha / beta / what? Let's explain it.

Reply:
```suggestion
- ApplySet-based pruning: An ApplySet is a server-side object (by default, a Secret) that kubectl can use to accurately and efficiently track set membership across `apply` operations. This mode was introduced in kubectl v1.27 as a replacement for allowlist-based pruning. It is in alpha and under active development in Kubernetes {{< skew currentVersion >}}.
```